### PR TITLE
Update debug branch for IPEX integration

### PR DESCRIPTION
**Pull request by kprokofi**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by kprokofi** [comment]
@eunwoosh , please take a look on updated ModelDebugger and simple training loop. However, I found differences in validation between full pipeline and this localized one

**Review by eunwoosh — COMMENTED** [review]
LGTM. Thanks for your update. I can refer your implementation.
It seems that learning rate isn't updated while training. Is this code just for simulating overall training code flow in simple way not for real training?

**Comment by kprokofi** [comment]
> LGTM. Thanks for your update. I can refer your implementation. It seems that learning rate isn't updated while training. Is this code just for simulating overall training code flow in simple way not for real training?

Yeah, but actually we can add learning rate scheduler as well.

**Review by eunwoosh — APPROVED** [review]
